File: basic/permutation/PermutationWithRepetition.py
import sys
input = sys.stdin.readline

# Read N (range of numbers) and M (length of sequence)
N, M = map(int, input().split())
sequence = [0] * M      # List to store the current sequence
result_lines = []       # List to accumulate the output lines

def dfs(depth):
    if depth == M:
        # Append the current sequence (converted to a space-separated string) to the result list
        result_lines.append(" ".join(map(str, sequence)))
        return
    for i in range(1, N + 1):
        sequence[depth] = i
        dfs(depth + 1)
        sequence[depth] = 0  # Reset the value

dfs(0)
print("\n".join(result_lines))

File: basic/permutation/PermutationWithRepetition.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const [N, M] = input[0].split(" ").map(Number);

let sequence = Array(M).fill(0); // Array to store the current sequence
let result = "";                 // String to accumulate the output

// DFS function: Selects numbers at the current depth
function dfs(depth) {
    if (depth === M) {
        result += sequence.join(" ") + "\n";
        return;
    }
    for (let i = 1; i <= N; i++) {
        sequence[depth] = i;
        dfs(depth + 1);
        sequence[depth] = 0; // Reset the value
    }
}

dfs(0);
console.log(result);
